lib/puppet/provider/user/directoryservice.rb

```diff
@@ -145,12 +145,12 @@ def self.generate_attribute_hash(input_hash)
       attribute_hash[:password] = '*'
     else
       embedded_binary_plist = get_embedded_binary_plist(attribute_hash[:shadowhashdata])
-      if embedded_binary_plist['SALTED-SHA512']
-        attribute_hash[:password] = get_salted_sha512(embedded_binary_plist)
-      else
+      if embedded_binary_plist['SALTED-SHA512-PBKDF2']
```

Comment:
On my 10.11.x machine, my user record's ShadowHashData contains both SALTED-SHA512-PBKDF2 and SALTED-SHA512 hash data; SALTED-SHA512-PBKDF2 is more secure, and preferred, so if it exists, return it, then fall through to SALTED-SHA512.

Actually, to expand a bit, the ShadowHashData had these: ["CRAM-MD5", "SALTED-SHA512-PBKDF2", "NT", "SRP-RFC5054-4096-SHA512-PBKDF2", "RECOVERABLE"]`

I expect puppet should eventually start supporting  "SRP-RFC5054-4096-SHA512-PBKDF2"